rubrik_cdm/data_management.py

```diff
@@ -1727,3 +1678,56 @@ def get_vsphere_vm(self, name=None, is_relic=None, effective_sla_domain_id=None,
         
         self.log("get_vsphere_vm: Get summary of all the VMs.")
         return self.get('v1', '/vmware/vm?{}'.format(query), timeout)
+
+    def get_vsphere_vm_snapshot(self, vm_name, timeout=15):  # pylint: ignore
+        """Retrieve summary information for the snapshots of a virtual machine.
+        Arguments:
+            vm_name {str} -- Name of the virtual machine.
+        Keyword Arguments:
+            timeout {int} -- The number of seconds to wait to establish a connection with the Rubrik cluster before returning a timeout error. (default: {15})
+        Returns:
+            dict -- The full response of `GET /v1/vmware/vm/{vm_id}/snapshot`
+        """
+
+        self.log("get_vsphere_vm_snapshot: Searching the Rubrik cluster for the vSphere VM '{}'.".format(vm_name))
+        vm_id = self.object_id(vm_name, 'vmware', timeout=timeout)
+
+        self.log("get_vsphere_vm_snapshot: Getting summary information for the snapshots of virtual machine {}".format(vm_id))
+        return self.get('v1', '/vmware/vm/{}/snapshot'.format(vm_id), timeout)
+
+    def get_vsphere_vm_details(self, vm_name, timeout=15):  # pylint: ignore
+        """Retrieve details for a virtual machine.
+        Arguments:
+            vm_name {str} -- Name of the virtual machine.       
+        Keyword Arguments:
+            timeout {int} -- The number of seconds to wait to establish a connection with the Rubrik cluster before returning a timeout error. (default: {15})
+        Returns:
+            dict -- The full response of `GET /v1/vmware/vm/{vm_id}`
+        """ 
+
+        self.log("get_vsphere_vm_details: Searching the Rubrik cluster for the vSphere VM '{}'.".format(vm_name))
+        vm_id = self.object_id(vm_name, 'vmware', timeout=timeout)
+
+        self.log("get_vsphere_vm_details: Getting details of virtual machine {}".format(vm_id))
+        return self.get('v1', '/vmware/vm/{}'.format(vm_id), timeout)
+
+    def get_vsphere_vm_file(self, vm_name, path=None, timeout=15):  # pylint: ignore
+        """Search for a file in the snapshots of a virtual machine. Specify the file by full path prefix or filename prefix.
+        Arguments:
+            vm_name {str} -- Name of the virtual machine.
+        Keyword Arguments:
+            timeout {int} -- The number of seconds to wait to establish a connection with the Rubrik cluster before returning a timeout error. (default: {15})
+            path {str} -- The path query. Use either a path prefix or a filename prefix.       
+        Returns:
+            dict -- The full response of `GET /v1/vmware/vm/{vm_id}/search?path={path}`
+        """
+
+        self.log("get_vsphere_vm_file: Searching the Rubrik cluster for the vSphere VM '{}'.".format(vm_name))
+        vm_id = self.object_id(vm_name, 'vmware', timeout=timeout)
+
+        if path == None:
+            raise InvalidParameterException('The path parameter argument must be provided')
```

Comment:
We can just make path a regular argument and it will be required automatically and we wont have to add a check.

i.e `def get_vsphere_vm_file(self, vm_name, path, timeout=15):`

Reply:
corrections committed